src/BTVNanoCommissioning/workflows/BTA_ttbar_producer.py

```diff
@@ -0,0 +1,527 @@
+        dataset = events.metadata["dataset"]
+        events = missing_branch(events)
+        shifts = []
+
+        if "JME" in self.SF_map.keys() and self.isJERC:
+            shifts = JME_shifts(
+                shifts, self.SF_map, events, self._campaign, isRealData, False, True
+            )
+        else:
+            shifts = [
+                ({"Jet": events.Jet, "MET": events.MET, "Muon": events.Muon}, None)
+            ]
+
+        return processor.accumulate(
+            self.process_shift(update(events, collections), name)
+            for collections, name in shifts
+        )
+
+    def process_shift(self, events, shift_name):
+        dataset = events.metadata["dataset"]
+        isRealData = not hasattr(events, "genWeight")
+        events = missing_branch(events)
+
+        # basic variables
+        basic_vars = {
+            "Run": events.run,
+            "Evt": events.event,
+            "LumiBlock": events.luminosityBlock,
+            "rho": events.fixedGridRhoFastjetAll,
+        }
+        if not isRealData:
+            basic_vars["nPU"] = events.Pileup.nPU
+            basic_vars["nPUtrue"] = events.Pileup.nTrueInt
+            basic_vars["pthat"] = events.Generator.binvar
+
+        ###########
+        #   HLT   #
+        ###########
+        triggers = [t[0] for t in BTA_ttbar_HLT_chns]
+
+        checkHLT = ak.Array([hasattr(events.HLT, _trig) for _trig in triggers])
+        if ak.all(checkHLT == False):
+            raise ValueError("HLT paths:", triggers, " are all invalid in", dataset)
+        elif ak.any(checkHLT == False):
+            print(
+                np.array(triggers)[~checkHLT],
+                " not exist in",
+                dataset,
+                ", please check!",
+            )
+
+        # pass_trigger dim: (n_evt, n_trig), append False for non-exist triggers
+        pass_trig = np.array(
+            [
+                getattr(events.HLT, _trig, np.zeros(len(events), dtype="bool"))
+                for _trig in triggers
+            ]
+        ).T
+
+        # get trigger pass for each channel (e, m, em)
+        pass_trig_chn = {
+            chn: np.zeros(len(events), dtype="bool")
+            for chn in ["e", "m", "ee", "mm", "em"]
+        }
+        for i, (trig, chn) in enumerate(
+            BTA_ttbar_HLT_chns
+        ):  # loop over e, m, em chanenl
+            pass_trig_chn[chn] |= pass_trig[:, i]
+
+        # convert to bitwise trigger (total number of triggers is less than 32)
+        basic_vars[f"ttbar_trigWord"] = to_bitwise_trigger(
+            pass_trig, ak.ArrayBuilder()
+        ).snapshot()[:, 0]
+
+        #########################
+        #  Lep/di-lep channels  #
+        #########################
+        # select offline electrons and muons (BTA TTbarSelectionProducer definition)
+        eles = events.Electron
+        eles = eles[
+            (eles.pt > 20)
+            & (abs(eles.deltaEtaSC + eles.eta) < 2.4)
```

Comment:
Looks like it, but why el_etaSC instead of el_eta itself? the SC eta is usually only needed to exclude the ECAL barrel-endcap transition region via 1.4442 < abs(el_etaSC) < 1.566 (not sure if the numbers are 100% correct), do we actually apply this cut somewhere in the repo?